XenModel/Actions/VM/VMMoveAction.cs

```diff
@@ -95,68 +96,77 @@ private void PopulateApiMethodsToRoleCheck()
 
         protected override void Run()

```

Comment:
I think I understand the intention around having the messages refer to the specific VM that is moving.
I'm a bit in the dark around the other changes, which appear to be:
1. Allowing errors in destroying VDIs to not stop the processing of others (and be reported later).
2. The removal of Cancel action code, is this handled in generic action handling code somewhere else these day?

I'll leave this one for the time being and look at other PRs, I probably just need to come back later when I've got my head back into it.

Reply:
1. The other changes were aimed at making the code a bit more efficient (for example, do not calculate or resolve anything before/unless it is needed) and simpler to understand (I'm trying to get rid of the base method BestEffort() and its passing around refs and delegates, because it is unnecessary complexity in the code; there are still a few instances of it that can be removed in future PRs). Otherwise, the changed code does the same as before: once the VM is copied, the VDIs at the old location should be destroyed, but if one fails, there is no need to stop and prevent the rest in the list from being destroyed, hence the late failure.
2.  In this particular case, catching and rethrowing the CancelException serves for showing a specialised "move cancelled" message. Generally, catching and re-throwing exceptions has some performance impact, so I thought it's not worth it since the base class does handle CancelException, albeit with a slightly more generic "cancelled" message. There are many other actions, particularly the ones more recently written, that prefer the latter.